sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication_async.py

```diff
@@ -23,14 +23,27 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
-from typing import TYPE_CHECKING
+from typing import TypeVar, Awaitable, Optional
 
-from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
+from azure.core.pipeline.policies import (
+    AsyncBearerTokenCredentialPolicy,
+    AsyncHTTPPolicy,
+)
+from azure.core.pipeline import PipelineRequest, PipelineResponse
 
-from ._authentication import _parse_claims_challenge
+from ._authentication import _parse_claims_challenge, _AuxiliaryAuthenticationPolicyBase
 
-if TYPE_CHECKING:
-    from azure.core.pipeline import PipelineRequest, PipelineResponse
+
+HTTPRequestType = TypeVar("HTTPRequestType")
+AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType")
+
+
+async def await_result(func, *args, **kwargs):
+    """If func returns an awaitable, await it."""
+    result = func(*args, **kwargs)
+    if hasattr(result, "__await__"):
```

Comment:
Why are we not using `inspect.isawaitable`? I don't believe perf is the reason here, correct?